crates/exex/exex/src/backfill.rs

```diff
@@ -185,41 +185,70 @@ where
         let chain = Chain::new(blocks, executor.finalize(), None);
         Ok(chain)
     }
+
+    /// Returns an iterator that executes each block in the range separately and yields the
+    /// resulting [`Block`] and [`ExecutionOutcome`].
+    pub fn into_single_blocks(
+        &self,
+    ) -> impl Iterator<Item = Result<(Block, ExecutionOutcome), BlockExecutionError>> + '_ {
+        self.range.clone().map(|block_number| {
+            let td = self
+                .provider
+                .header_td_by_number(block_number)?
+                .ok_or_else(|| ProviderError::HeaderNotFound(block_number.into()))?;
+
+            // Fetch the block with senders for execution.
+            let block_with_senders = self
+                .provider
+                .block_with_senders(block_number.into(), TransactionVariant::WithHash)?
+                .ok_or_else(|| ProviderError::HeaderNotFound(block_number.into()))?;
+
+            // Configure the executor to use the previous block's state.
+            let mut executor = self.executor.batch_executor(
+                StateProviderDatabase::new(
+                    self.provider.history_by_block_number(block_number.saturating_sub(1))?,
+                ),
+                self.prune_modes.clone(),
+            );
```

Comment:
no need for a batch executor here, can just do `self.executor.executor()` and call the `execute()` method that instantly returns the result without an additional call to `finalize`

Reply:
using the `executor()` means that we receive a `BlockExecutionOutput`, do you want to convert this to an `ExecutionOutcome` or just keep the `BlockExecutionOutput`?